adapters/repos/db/lsmkv/strategies_map_sorted_merger.go

```diff
@@ -92,6 +111,42 @@ func (s *sortedMapMerger) init(segments [][]MapPair) error {
 	return nil
 }
 
+// reset can be manually called if sharing allocated state is desired, such as
+// with .doKeepTombstonesReusable()
+func (s *sortedMapMerger) reset(segments [][]MapPair) error {
```

Comment:
this is very easy to understand with these comments

Reply:
Thanks 😍